### PR TITLE
added pythonPackages.tensorboardX

### PR DESCRIPTION
**Pull request by akamaus**
###### Motivation for this change

Added an expression for tensorboardX, a standalone library for logging data for subsequent visualization by tensorboard.

###### Things done

- Built on platform(s)
   - [ X] NixOS
   - [X ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by FRidh — CHANGES_REQUESTED** [review]
Please use the full PR template, it's there for a reason.

**Comment by akamaus** [comment]
@FRidh all done

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4933,6 +4933,8 @@ in {
 
   suseapi = callPackage ../development/python-modules/suseapi { };
 
+  tensorboardX = callPackage ../development/python-modules/tensorboardX { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: attribute name and filename needs to be the normalized name, so all lowercase

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.tensorboardX

**Comment by akamaus** [comment]
@FRidh renamed to lowercase.

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by FRidh** [comment]
@GrahamcOfBorg eval

**Review by FRidh — COMMENTED** [review]
Looks ok

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{lib, fetchFromGitHub, buildPythonPackage, matplotlib, numpy, pillow, pytorch, protobuf, six, pytest, torchvision }:
+
+buildPythonPackage rec {
+  pname = "tensorboardx";
+  version = "1.6";
```
Comment (mmahut) [inline pkgs/development/python-modules/tensorboardx/default.nix]: Can we update this to the latest version?

Comment (akamaus) [reply pkgs/development/python-modules/tensorboardx/default.nix]: @mmahut, done. Library became even larger. I had to disable some tests. Most failed for spurious reasons, like "44100" vs "44100.0" in audioclip in serialized prototxt.

File: pkgs/development/python-modules/crc32c/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  version = "1.7";
+
+  src = fetchFromGitHub {
+    owner = "ICRAR";
+    repo = "crc32c";
+    rev = "v${version}";
+    sha256 = "0c3wwxwkm7ax6lw5w053iw16izaqd6fq6j5185fnv8mg6liflaad";
+  };
+
+  checkInputs = [ ];
```
Comment (mmahut) [inline pkgs/development/python-modules/crc32c/default.nix]: Any reason to keep these around if they are empty?

File: pkgs/development/python-modules/crc32c/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  propagatedBuildInputs = [ ];
+
+  meta = {
```
Comment (mmahut) [inline pkgs/development/python-modules/crc32c/default.nix]: please include ```platform```.

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    substituteInPlace tests/test_summary.py --replace test_audio _skip_test_audio
+    substituteInPlace tests/test_summary.py --replace test_histogram_auto _skip_histogram_auto
+    substituteInPlace tests/test_summary.py --replace test_histogram_doane  _skip_test_histogram_doane
+    substituteInPlace tests/test_summary.py --replace test_histogram_fd _skip_test_histogram_fd
+
+    substituteInPlace tests/test_visdom.py --replace test_TorchVis _skip_test_TorchVis
+    substituteInPlace tests/test_onnx_graph.py --replace test_onnx_graph _skip_test_onnx_graph
+    '';
+
+  meta = {
```
Comment (mmahut) [inline pkgs/development/python-modules/tensorboardx/default.nix]: Please include ```platforms```.

File: pkgs/development/python-modules/crc32c/default.nix (hunk 3 above)
Comment (akamaus) [reply pkgs/development/python-modules/crc32c/default.nix]: @mmahut , right, removed.

File: pkgs/development/python-modules/crc32c/default.nix (hunk 4 above)
Comment (akamaus) [reply pkgs/development/python-modules/crc32c/default.nix]: @mmahut  done

File: pkgs/development/python-modules/tensorboardx/default.nix (hunk 5 above)
Comment (akamaus) [reply pkgs/development/python-modules/tensorboardx/default.nix]: @mmahut done, borg is finally pleased, too.

**Comment by akamaus** [comment]
@FRidh, please have a look.

**Comment by akamaus** [comment]
Updated PR to latest tensorboardX-2.0, apparently, previosly failed tests were fixed upstream, so now they all pass (besides two, one for Visdom integration which is missing from nixpkgs and one trying download something with wget. All builds and works nicely for me.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -283,6 +283,11 @@
     githubId = 273837;
     name = "Mateusz Czapliński";
   };
+  akamaus = {
+    email = "dmitryvyal@gmail.com";
+    github = "akamaus";
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:288]:
```suggestion
    github = "akamaus";
    githubId = 58955;
```
https://gist.github.com/GrahamcOfBorg/e5c60cefeb6835923d8e2041bb22fa0a

**Review by jonringer — COMMENTED** [review]
I think it also makes sense to squash all of the tensorboard commits, and have just a single `python3Packages.tensorboardx: init at 2.0`.

From the perspective master branch, the meaningful change is that a package was added, the intermediate bumps are not very consequential (at least to me).

**Comment by Mic92** [comment]
I think @mweinelt has an interest in getting this merged.

**Comment by akamaus** [comment]
@Mic92 commits squashed, githubId added.

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{boto3, buildPythonPackage, crc32c, fetchFromGitHub, lib, matplotlib, moto, numpy, pillow, pytorch, protobuf, six, pytest, tensorflow-tensorboard, torchvision }:
```
Comment (mweinelt) [inline pkgs/development/python-modules/tensorboardx/default.nix:1]: Can you wrap this line? It is quite a bit too long to properly read on GitHub.

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  propagatedBuildInputs = [ numpy protobuf six ];
+
+  postPatch=''
+    substituteInPlace tests/test_visdom.py --replace test_TorchVis _skip_test_TorchVis
+    substituteInPlace tests/test_onnx_graph.py --replace test_onnx_graph _skip_test_onnx_graph
+    '';
+
+  meta = {
+    description = "Library for writing tensorboard-compatible logs";
+    homepage = https://github.com/lanpa/tensorboard-pytorch;
```
Comment (mweinelt) [inline pkgs/development/python-modules/tensorboardx/default.nix:25]: The URL needs to be quoted as per RFC 45: https://github.com/NixOS/rfcs/blob/master/rfcs/0045-deprecate-url-syntax.md

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  checkInputs = [ boto3 crc32c matplotlib moto pillow pytorch pytest tensorflow-tensorboard torchvision ];
+
+  propagatedBuildInputs = [ numpy protobuf six ];
+
+  postPatch=''
+    substituteInPlace tests/test_visdom.py --replace test_TorchVis _skip_test_TorchVis
+    substituteInPlace tests/test_onnx_graph.py --replace test_onnx_graph _skip_test_onnx_graph
+    '';
+
+  meta = {
```
Comment (mweinelt) [inline pkgs/development/python-modules/tensorboardx/default.nix:23]:
```suggestion
  meta = with lib; {
```

And then cut off `lib.` on `license`, `maintainers` and `platforms`.

File: pkgs/development/python-modules/tensorboardx/default.nix [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  checkInputs = [ boto3 crc32c matplotlib moto pillow pytorch pytest tensorflow-tensorboard torchvision ];
+
+  propagatedBuildInputs = [ numpy protobuf six ];
+
+  postPatch=''
+    substituteInPlace tests/test_visdom.py --replace test_TorchVis _skip_test_TorchVis
+    substituteInPlace tests/test_onnx_graph.py --replace test_onnx_graph _skip_test_onnx_graph
+    '';
```
Comment (mweinelt) [inline pkgs/development/python-modules/tensorboardx/default.nix:21]: The indentation here seems off.

**Review by mweinelt — CHANGES_REQUESTED** [review]
That's a great start, I only added some style remarks.

**Comment by akamaus** [comment]
@mweinelt, thanks for tips for fixing style. All done.

**Comment by Mic92** [comment]
@GrahamcOfBorg build python3.pkgs.tensorboardX

**Comment by Mic92** [comment]
@GrahamcOfBorg build python3.pkgs.tensorboardx

**Pull request merged**